dotnet/src/IntegrationTests/Connectors/Memory/Postgres/PostgresVectorStoreRecordCollectionTests.cs

```diff
@@ -0,0 +1,525 @@
+        if (await sut.CollectionExistsAsync())
+        {
+            await sut.DeleteCollectionAsync();
+        }
+
+        await sut.CreateCollectionAsync();
+
+        var writtenHotel1 = new PostgresHotel<int> { HotelId = 1, HotelName = "Hotel 1", HotelCode = 1, ParkingIncluded = true, HotelRating = 4.5f, Tags = ["tag1", "tag2"] };
+        var writtenHotel2 = new PostgresHotel<int> { HotelId = 2, HotelName = "Hotel 2", HotelCode = 2, ParkingIncluded = false, HotelRating = 2.5f, ListInts = [1, 2] };
+
+        try
+        {
+            // Act
+
+            await sut.UpsertAsync(writtenHotel1);
+
+            await sut.UpsertAsync(writtenHotel2);
+
+            var fetchedHotel1 = await sut.GetAsync(1);
+            var fetchedHotel2 = await sut.GetAsync(2);
+
+            // Assert
+            Assert.NotNull(fetchedHotel1);
+            Assert.Equal(1, fetchedHotel1!.HotelId);
+            Assert.Equal("Hotel 1", fetchedHotel1!.HotelName);
+            Assert.Equal(1, fetchedHotel1!.HotelCode);
+            Assert.True(fetchedHotel1!.ParkingIncluded);
+            Assert.Equal(4.5f, fetchedHotel1!.HotelRating);
+            Assert.NotNull(fetchedHotel1!.Tags);
+            Assert.Equal(2, fetchedHotel1!.Tags!.Count);
+            Assert.Equal("tag1", fetchedHotel1!.Tags![0]);
+            Assert.Equal("tag2", fetchedHotel1!.Tags![1]);
+            Assert.Null(fetchedHotel1!.ListInts);
+            Assert.Equal(TruncateMilliseconds(fetchedHotel1.CreatedAt), TruncateMilliseconds(writtenHotel1.CreatedAt));
+            Assert.Equal(TruncateMilliseconds(fetchedHotel1.UpdatedAt), TruncateMilliseconds(writtenHotel1.UpdatedAt));
+
+            Assert.NotNull(fetchedHotel2);
+            Assert.Equal(2, fetchedHotel2!.HotelId);
+            Assert.Equal("Hotel 2", fetchedHotel2!.HotelName);
+            Assert.Equal(2, fetchedHotel2!.HotelCode);
+            Assert.False(fetchedHotel2!.ParkingIncluded);
+            Assert.Equal(2.5f, fetchedHotel2!.HotelRating);
+            Assert.NotNull(fetchedHotel2!.Tags);
+            Assert.Empty(fetchedHotel2!.Tags);
+            Assert.NotNull(fetchedHotel2!.ListInts);
+            Assert.Equal(2, fetchedHotel2!.ListInts!.Count);
+            Assert.Equal(1, fetchedHotel2!.ListInts![0]);
+            Assert.Equal(2, fetchedHotel2!.ListInts![1]);
+            Assert.Equal(TruncateMilliseconds(fetchedHotel2.CreatedAt), TruncateMilliseconds(writtenHotel2.CreatedAt));
+            Assert.Equal(TruncateMilliseconds(fetchedHotel2.UpdatedAt), TruncateMilliseconds(writtenHotel2.UpdatedAt));
+        }
+        finally
+        {
+            // Cleanup
+            await sut.DeleteCollectionAsync();
+        }
+    }
+
+    [Theory]
+    [InlineData(typeof(short), (short)3)]
+    [InlineData(typeof(int), 5)]
+    [InlineData(typeof(long), 7L)]
+    [InlineData(typeof(string), "key1")]
+    [InlineData(typeof(Guid), null)]
+    public async Task ItCanGetAndDeleteRecordAsync(Type idType, object? key)
```

Comment:
It is also possible to make test methods generic, e.g. `public async Task ItCanGetAndDeleteRecordAsync<TKey>(Type idType, TKey? key)`, which means you can use TKey in the method and you don't have to use dynamic.